include/wil/filesystem.h

```diff
@@ -19,6 +19,7 @@
 #include <combaseapi.h> // Needed for CoTaskMemFree() used in output of some helpers.
 #include <winbase.h> // LocalAlloc
 #include <PathCch.h>
+#include <cassert>

```

Comment:
some systems use a different assert macro controlled by build system specific macros. notice how there are no other uses of cassert in wil.

Reply:
You are right, but using assert (or such kind of kindred statements) is for a debug mode as You know. Because of this, I already mentioned about it in my 1st explanation, about using an "if statement" to prevent of sending the "nullptr" (since the "wcsncmp" function doesn't validate its parameters for other situations beside this via giving an exception):

inline bool is_extended_length_path(_In_ PCWSTR path)
{
        if(s)
            return wcsncmp(path, L"\\\\?\\", 4) == 0;
}